src/firefly/js/tables/TableUtil.js

```diff
@@ -999,6 +999,42 @@ export function setHlRowByRowIdx(nreq, tableModel) {
     }
 }
 
+
+/**
+ *
+ * @param {TableModel|String} tableOrId - parameters accepts the table model or tha table id
+ * @param {number} rowIdx - rowIdx to check
+ * @param {number} [hlRowIdx] - highlighted row index, defaults to tableModel.highlightedRow
+ * @return {boolean} true if it is sub-highlight
+ */
+export function isSubHighlightRow(tableOrId, rowIdx, hlRowIdx) {
+    const tableModel = getTM(tableOrId);
+    if (!tableModel?.tableMeta) return false;
+    const relatedCols = tableModel.tableMeta['tbl.relatedCols'];
+    if (!relatedCols) return false;
+
+    const highlightedRow= hlRowIdx ?? tableModel.highlightedRow;
+
+    const relatedColsHL = relatedCols
+        .split(',')
+        .map((cname) => getCellValue(tableModel, highlightedRow, cname?.trim()))
+        .join('|');
+
+    const cVal = relatedCols
+        .split(',')
+        .map((cname) => getCellValue(tableModel, rowIdx, cname?.trim()))
+        .join('|');
```

Comment:
this chained string manipulation can be defined as an inline function with row as a param, and then can be reused in both cases.